test/wodin.test.ts

```diff
@@ -132,3 +133,92 @@ describe("can filter", () => {
         expect(filterIndex(t, times)).toStrictEqual([31, 78]);
     });
 });
+
+describe("can run batch", () => {
+    const allTimes = {
+        mode: TimeMode.Grid as const,
+        tStart: 0,
+        tEnd: 10,
+        nPoints: Infinity
+    };
+    it("can run", () => {
+        const tEnd = 10;
+        const dt = 0.1;
+        const nParticles = 7;
+        const pars = {
+            base: {n: 1, sd: 0},
+            name: "sd",
+            values: [0, 1, 10, 100]
+        };
+        const res = batchRunDiscrete(models.Walk, pars, 0, tEnd, dt, nParticles);
+        res.run();
+        expect(res.solutions.length).toBe(4);
+        expect(res.errors).toStrictEqual([]);
+        const traces0 = res.solutions[0](allTimes);
+        expect(traces0.x).toStrictEqual(seq(0, 100).map((x) => x * dt));
+        expect(traces0.values).toStrictEqual(
+            [{ name: "x", y: Array(101).fill(0) }]);
+        const traces1 = res.solutions[1](allTimes);
+        expect(traces1.values.length).toBe(1);
+        expect(traces1.values[0].name).toBe("x");
+        expect(traces1.values[0].y.slice(1).every((x) => x !== 0)).toBe(true);
+
+        const end = res.valueAtTime(10);
+        expect(end.x).toStrictEqual(pars.values);
+        expect(end.values.length).toBe(1);
+        expect(end.values[0].name).toBe("x");
+        expect(end.values[0].y).toStrictEqual(res.solutions.map((el) => el(allTimes).values[0].y[100]));
+
+        const max = res.extreme("yMax");
+        expect(max.x).toStrictEqual(pars.values);
+        expect(max.values.length).toBe(1);
+        expect(max.values[0].name).toBe("x");
+        expect(max.values[0].y).toStrictEqual(
+            res.solutions.map((el) => Math.max(...el(allTimes).values[0].y)));
+    });
+
+    it("can report on near miss values", () => {
```

Comment:
I'm not really sure what 'near miss' means here, or how this tests for it.

Reply:
```
        // Unlike odin, if the user asks for a specific time it might
        // not be present due to floating point differences; check
        // that if they're close to a time we get the right time back.
```
added some explanation here now